packages/features/users/components/UserTable/UserListTable.tsx

```diff
@@ -0,0 +1,353 @@
+  const { data: session } = useSession();
+  const { data: currentMembership } = trpc.viewer.organizations.listCurrent.useQuery();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { t } = useLocale();
+  const utils = trpc.useContext();
+
+  const { data, isLoading, fetchNextPage, isFetching } =
+    trpc.viewer.organizations.listMembers.useInfiniteQuery(
+      {
+        limit: 10,
+      },
+      {
+        getNextPageParam: (lastPage) => lastPage.nextCursor,
+        keepPreviousData: true,
+      }
+    );
+
+  const adminOrOwner = currentMembership?.user.role === "ADMIN" || currentMembership?.user.role === "OWNER";
+
+  const memorisedColumns = useMemo(() => {
+    const permissions = {
+      canEdit: adminOrOwner,
+      canRemove: adminOrOwner,
+      canImpersonate: false,
+    };
+    const cols: ColumnDef<User>[] = [
+      // Disabling select for this PR: Will work on actions etc in a follow up
+      // {
+      //   id: "select",
+      //   header: ({ table }) => (
+      //     <Checkbox
+      //       checked={table.getIsAllPageRowsSelected()}
+      //       onCheckedChange={(value) => table.toggleAllPageRowsSelected(!!value)}
+      //       aria-label="Select all"
+      //       className="translate-y-[2px]"
+      //     />
+      //   ),
+      //   cell: ({ row }) => (
+      //     <Checkbox
+      //       checked={row.getIsSelected()}
+      //       onCheckedChange={(value) => row.toggleSelected(!!value)}
+      //       aria-label="Select row"
+      //       className="translate-y-[2px]"
+      //     />
+      //   ),
+      // },
+      {
+        id: "member",
+        accessorFn: (data) => data.email,
+        header: "Member",
+        cell: ({ row }) => {
+          const { username, email } = row.original;
+          return (
+            <div className="flex items-center gap-2">
+              <Avatar size="sm" alt={username || email} imageSrc={`/${username}/avatar.png`} />
+              <div className="">
+                <div className="text-emphasis text-sm font-medium leading-none">
+                  {username || "No username"}
+                </div>
+                <div className="text-subtle text-sm leading-none">{email}</div>
+              </div>
+            </div>
+          );
+        },
+        filterFn: (rows, id, filterValue) => {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore Weird typing issue
+          return rows.getValue(id).includes(filterValue);
+        },
+      },
+      {
+        id: "role",
+        accessorFn: (data) => data.role,
+        header: "Role",
+        cell: ({ row, table }) => {
+          const { role } = row.original;
+          return (
+            <Badge
+              variant={role === "MEMBER" ? "gray" : "blue"}
+              onClick={() => {
+                table.getColumn("role")?.setFilterValue(role);
```

Comment:
Allow drill down upon clicking a badge -> clicking ADMIN will filter to all ADMINs in the loaded dataset